pkg/replication/replicator.go

```diff
@@ -110,6 +117,58 @@ func (r replicator) InitializeReplication(
 	return replication, errCh, nil
 }
 
+// Initializes a replication after performing some initial validation.
+// Validation errors are returned immediately, and asynchronous errors are
+// passed on the returned channel
+func (r replicator) InitializeReplicationWithCheck(
+	overrideLock bool,
+	ignoreControllers bool,
+	concurrentRealityRequests int,
+	checkPreparers bool,
+) (Replication, chan error, error) {
```

Comment:
This function is copy pasted from `InitializeReplication`. I'd like to see `InitializeReplication` call this function with a checkPreparers argument of  `true` so that we can reduce code duplication.